go/vt/wrangler/traffic_switcher.go

```diff
@@ -1759,3 +1819,43 @@ func reverseName(workflow string) string {
 	}
 	return workflow + reverse
 }
+
+func (ts *trafficSwitcher) addParticipatingTablesToKeyspace(ctx context.Context, keyspace, tableSpecs string) error {
+	var err error
+	var vschema *vschemapb.Keyspace
+	vschema, err = ts.wr.ts.GetVSchema(ctx, keyspace)
+	if err != nil {
+		return err
+	}
+	if vschema == nil {
+		return fmt.Errorf("no vschema found for keyspace %s", keyspace)
+	}
+	if strings.HasPrefix(tableSpecs, "{") {
+		if vschema.Tables == nil {
+			vschema.Tables = make(map[string]*vschemapb.Table)
+		}
+		wrap := fmt.Sprintf(`{"tables": %s}`, tableSpecs)
+		ks := &vschemapb.Keyspace{}
+		if err := json2.Unmarshal([]byte(wrap), ks); err != nil {
+			return err
+		}
+		if err != nil {
+			return err
+		}
+		for table, vtab := range ks.Tables {
+			vschema.Tables[table] = vtab
+		}
+	} else {
+		if !vschema.Sharded {
+			if vschema.Tables == nil {
+				vschema.Tables = make(map[string]*vschemapb.Table)
+			}
+			for _, table := range ts.tables {
+				vschema.Tables[table] = &vschemapb.Table{}
+			}
+		} else {
+			return fmt.Errorf("no sharded vschema was provided, so you will need to update the vschema of the target manually for the moved tables")
+		}
+	}
+	return ts.wr.ts.SaveVSchema(ctx, keyspace, vschema)
```

Comment:
I feel like there's too many `if-else` paths in this function, and it confuses me. Any way to reorganize it?

Reply:
reorganized: it does read better now!